service/src/main/java/tds/assessment/services/impl/AssessmentAssembler.java

```diff
@@ -53,29 +53,29 @@ static void assemble(final Assessment assessment,
             .map(ItemProperty::getValue)
             .collect(Collectors.toSet());
         assessment.setLanguageCodes(languages);
-        
+
         // Update items with their appropriate item properties
         for (Item item : items) {
             List<ItemProperty> properties = itemProperties.stream()
                 .filter(ip -> ip.getItemId().equals(item.getId()))
                 .collect(Collectors.toList());
-            
+
             item.setItemProperties(properties);
         }
-        
+
         // FIXED FORM SEGMENTS - If forms were supplied, wire up the items to their appropriate form
         // RULE:  The items in the form should be sorted by their formPosition in ascending order
         if (forms.size() > 0) {
             for (Form form : forms) {
                 List<Item> itemsForThisForm = items.stream()
-                    .filter(i -> i.getFormKeys() != null
-                        && i.getFormKeys().contains(form.getKey())
+                    .filter(i -> i.getFormKey() != null
```

Comment:
Since we're not collapsing the items in the repository we simply iterate over all of them and assign them to the proper form.